src/output/plugins/OSXOutputPlugin.cxx

```diff
@@ -688,7 +680,44 @@ osx_render(void *vdata,
 {
 	OSXOutput *od = (OSXOutput *) vdata;
 
-	int count = in_number_frames * od->asbd.mBytesPerFrame;
+	if (od->cancel) {
+		od->ring_buffer->reset();
```

Comment:
This method is not thread-safe. Concurrent `Play()` calls may wreak havoc.

Reply:
that's why it's wrapped in atomic variable cancel. If cancel is true, Play() will never write to the ring buffer.

Reply:
I'll mark this as resolved.